core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -175,8 +182,7 @@ protected void refreshFromMetadataLocation(String newLocation, Predicate<Excepti
           .retry(numRetries).exponentialBackoff(100, 5000, 600000, 4.0 /* 100, 400, 1600, ... */)
           .throwFailureWhenFinished()
           .shouldRetryTest(shouldRetry)
-          .run(metadataLocation -> newMetadata.set(
-              TableMetadataParser.read(io(), metadataLocation)));
+          .run(metadataLocation -> newMetadata.set(metadataLoader.apply(metadataLocation)));
```

Comment:
I'm just catching up on this PR, sorry to be late.

I'm curious why this uses a `Function` passed into refresh rather than defining a load metadata method. The function passed in from `NessieTableOperations` is actually a method reference, so couldn't this just be a call to that method, `loadTableMetadata`?

```java
    .run(metadataLocation -> loadTableMetadata(metadataLocation));

...

  public TableMetadata loadTableMetadata(String metadataLocation) {
    return TableMetadataParser.read(io(), metadataLocation);
  }
```